amr-wind/wind_energy/actuator/disk/disk_spreading.H

```diff
@@ -0,0 +1,172 @@
+#ifndef DISK_SPREADING_H_
+#define DISK_SPREADING_H_
+
+#include "amr-wind/wind_energy/actuator/actuator_utils.H"
+#include "amr-wind/core/FieldRepo.H"
+
+namespace amr_wind {
+namespace actuator {
+namespace ops {
+
+/**
+ * @brief  A collection of spreading functions
+ * This class allows for polymorphic spreading functions.
+ * The concept is based on Impossibly Fast C++ Delegates proposed here:
+ * https://www.codeproject.com/Articles/11015/The-Impossibly-Fast-C-Delegates
+ *
+ * @tparam T
+ */
+template <typename T>
+class SpreadingFunction
+{
+public:
+    void operator()(
+        const T& actObj,
+        const int lev,
+        const amrex::MFIter& mfi,
+        const amrex::Geometry& geom)
+    {
+        (this->*m_function)(actObj, lev, mfi, geom);
+    }
+
+    SpreadingFunction(const SpreadingFunction&) = delete;
+    void operator=(const SpreadingFunction&) = delete;
+
+    void (SpreadingFunction::*m_function)(
+        const T& actObj,
+        const int,
+        const amrex::MFIter&,
+        const amrex::Geometry&);
+
+    void uniform_gaussian_spreading(
+        const T& actObj,
+        const int lev,
+        const amrex::MFIter& mfi,
+        const amrex::Geometry& geom)
+    {
+        const auto& bx = mfi.tilebox();
+        const auto& sarr = actObj.m_act_src(lev).array(mfi);
+        const auto& problo = geom.ProbLoArray();
+        const auto& dx = geom.CellSizeArray();
+
+        const auto& data = actObj.m_data.meta();
+
+        const vs::Vector epsilon = vs::Vector::one() * data.epsilon;
+        const vs::Vector m_normal(data.normal_vec);
+        const vs::Vector m_origin(data.center);
```

Comment:
unused variable? m_origin and m_coplanar

Reply:
ah nice! yeah I changed the algorithm so we don't need those. I'll remove them now